clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -596,6 +606,8 @@ public void handleResponse(AbstractResponse response) {
                 reenqueue();
             } else if (error == Errors.COORDINATOR_LOAD_IN_PROGRESS || error == Errors.CONCURRENT_TRANSACTIONS) {
                 reenqueue();
+            } else if (error == Errors.CLUSTER_AUTHORIZATION_FAILED) {
+                fatal(new ClusterAuthorizationException("The producer is not authorized to generate a producerId for idempotence"));
```

Comment:
As mentioned before, we should be using the `Errors.CLUSTER_AUTHORIZATION_FAILED.exception()` here so that the messages are consistent and can be updated consistently in the future.